src/Azure/Orleans.GrainDirectory.AzureStorage/AzureTableGrainDirectory.cs

```diff
@@ -72,21 +72,7 @@ public async Task<GrainAddress> Register(GrainAddress address)
             var entry = GrainDirectoryEntity.FromGrainAddress(this.clusterId, address);
             var result = await this.tableDataManager.InsertTableEntryAsync(entry);
             // Possible race condition?
-
-            if (result.isSuccess)
-            {
-                return address;
-            }
-
-            var lookupResult = await Lookup(address.GrainId);
-            if (lookupResult == null)
-            {
-                // It's not replicated to secondary yet. Retry after 100ms.
-                await Task.Delay(100);
-                lookupResult = await Lookup(address.GrainId);
-            }
-
-            return lookupResult;
+            return result.isSuccess ? address : await Lookup(address.GrainId);
```

Comment:
yes, I added the retry here for potential replication delay because we are using multi-region cosmosdb azure table. let's keep the retry

---
In reply to: [572315254](https://github.com/juyuz-dev/orleans/pull/7#discussion_r572315254) [](ancestors = 572315254)